packages/next-on-pages/src/buildApplication/processVercelFunctions/invalidFunctions.ts

```diff
@@ -55,28 +59,48 @@ export async function checkInvalidFunctions(
  *
  * @param collectedFunctions Collected functions from the Vercel build output.
  */
-async function tryToFixNotFoundRoute(
-	collectedFunctions: CollectedFunctions,
-): Promise<void> {
-	const functionsDir = resolve('.vercel', 'output', 'functions');
-	const notFoundDir = join(functionsDir, '_not-found.func');
-	const errorDir = join(functionsDir, '_error.func');
+async function tryToFixNotFoundRoute({
+	invalidFunctions,
+	ignoredFunctions,
+}: CollectedFunctions): Promise<void> {
+	for (const [fullPath, fnInfo] of invalidFunctions.entries()) {
+		const notFoundFuncName = '/_not-found.func';
+		const errorFuncName = '/_error.func';
 
-	const invalidNotFound = collectedFunctions.invalidFunctions.get(notFoundDir);
-	const invalidError = collectedFunctions.invalidFunctions.get(errorDir);
+		const invalidNotFound = fullPath.endsWith(notFoundFuncName);
```

Comment:
I _think_ that the `_not-found.func` hack is for not-found functions at the root, so I'm not sure if checking this with simply `endsWith` is correct.

I have to have a think about it 🤔

Reply:
Well, when using a base path, it creates one inside the base path, so....

Reply:
but I think that once we were trying to be smart about it and tried to detect when a base path was being used? (but checking all the routes and see if they have the same prefix?), did we scrap that? if not, could it be used here? :thinking:

Reply:
> but I think that once we were trying to be smart about it and tried to detect when a base path was being used? (but checking all the routes and see if they have the same prefix?), did we scrap that? if not, could it be used here? 🤔

i'm not sure, i can't remember that... even if we did do that though, it might fail due to invalid functions being generated outside the base path

Reply:
ok let's go with this, I feel like the risk of issues is hopefully pretty low :slightly_smiling_face: